src/expenses/services/expense.service.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+let expenses = [];
+
+const clearState = () => {
+  expenses = [];
+};
+
+const getAll = (userId, categories, from, to) => {
+  let newExpences = [...expenses];
+
+  if (userId) {
+    newExpences = newExpences.filter(
+      expence => expence.userId === userId
+    );
+  }
+
+  if (categories) {
+    newExpences = newExpences.filter(
+      expense => categories
+        .includes(expense.category)
+    );
+  }
+
+  if (from) {
+    newExpences = newExpences.filter(
+      expense => from <= expense.spentAt
+    );
+  }
+
+  if (to) {
+    newExpences = newExpences.filter(
+      expense => to >= expense.spentAt
+    );
+  }
+
+  return newExpences;
+};
+
+const getById = (id) => {
+  return expenses.find(expense => expense.id === id) || null;
+};
+
+const create = (
+  userId,
+  spentAt,
+  title,
+  amount,
+  category,
+  note,) => {
+  const expense = {
+    id: expenses.reduce((acc, item) => {
+      return acc > item.id ? acc : item.id;
+    }, 0) + 1,
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  expenses.push(expense);
+
+  return expense;
+};
+
+const update = (id, userId, spentAt, title, amount, category, note) => {
+  const expense = getById(+id);
+
+  if (userId !== undefined) {
```

Comment:
I think you don't need compare with undefined you can write (!userId)

Reply:
If you write `if (userId){} ` it would give the same result for undefined, null, and 0, but it should return true for undefined only. If userId is 0 or null, then this property should be updated. If userId is undefined - it shouldn't.